src/foam/nanos/menu/Menu.js

```diff
@@ -211,8 +211,16 @@
     {
       name: 'authorizeOnRead',
       javaCode: `
-        if ( ! getAuthenticate() ) return;
         AuthService auth = (AuthService) x.get("auth");
+        boolean unauthenticated = false;
+        try {
+          var subject = auth.getCurrentSubject(x);
+          if ( subject == null || auth.isUserAnonymous(x, subject.getUser().getId()) ) 
+            unauthenticated = true;
+        } catch(foam.nanos.auth.AuthenticationException e) {
+          unauthenticated = true;
```

Comment:
Can you explain this part?

Reply:
we only want to skip authentication for unauthenticated menus when there's no user in the session or only anonymous user in the session

Reply:
authentcation exception is because getcurrentsubject will throw authenticationexception when session is null or session user is null